### ISSUE TITLE
option to remove BrewTestBot checked formulae from brew livecheck output

### ISSUE THREAD

**Issue by eleanordoesntcode** [Homebrew/brew#18640]
### Verification

- [X] This issue's title and/or description do not reference a single formula e.g. `brew install wget`. If they do, open an issue at https://github.com/Homebrew/homebrew-core/issues/new/choose instead.

### Provide a detailed description of the proposed feature

An option to remove formulaes BrewTestBot check automatically from the output of `brew livecheck`. The command could verify if the formulae is in the list of formulaes that are checked by BrewTestBot before even sending a network request, to preserve on time wasted on checking formulaes that are automatically updated.

### What is the motivation for the feature?

This would preserve time on contributors like me, who want to help updating formulaes by running `brew livecheck --newer-only --eval-all` but don't want to spend time checking if the formulae is handled by BrewTestBot, since 80-90% of the formulaes outputed by this command are in the list.

### How will the feature be relevant to at least 90% of Homebrew users?

It won't be relevant to anybody who doesn't contribute to Homebrew, but for those who do, it would save quite some time.

### What alternatives to the feature have been considered?

A bash script can be made to only print a line of `brew livecheck` if the first word isn't in the BrewTestBot checked formulae list, but this is hacky and it might break.

**Comment by gromgit** [comment]
By "BrewTestBot checked formulae", do you mean _autobumped_ formula? If so, there's a one-liner for that:
```
comm -23 <(brew formulae | grep -v /) $(brew --repo homebrew/core)/.github/autobump.txt | xargs brew livecheck --newer-only
```

**Comment by MikeMcQuaid** [comment]
> This would preserve time on contributors like me, who want to help updating formulaes by running `brew livecheck --newer-only --eval-all` but don't want to spend time checking if the formulae is handled by BrewTestBot, since 80-90% of the formulaes outputed by this command are in the list.

Just to confirm I understand @eleanordoesntcode: is it that this command will still attempt to check the livecheck status for formulae it later complains about being handled by BrewTestBot and you'd rather they were skipped entirely?

If so: it kinda feels like this should be the default behaviour for end-users and we should instead be passing an argument/environment variable when running this in CI or in local live check testing to do otherwise.

**Comment by eleanordoesntcode** [comment]
> By "BrewTestBot checked formulae", do you mean _autobumped_ formula? If so, there's a one-liner to avoid live checking them at all:
> 
> ```
> comm -23 <(brew formulae | grep -v /) $(brew --repo homebrew/core)/.github/autobump.txt | xargs brew livecheck --formulae --newer-only
> ```

I did mean that yes.
Thanks for the one liner, but I'd prefer if it was directly built in instead of having to resort to this. It'd make the whole thing faster if it didn't check the BrewTestBot autobumped formulaes.

**Comment by eleanordoesntcode** [comment]
> > This would preserve time on contributors like me, who want to help updating formulaes by running `brew livecheck --newer-only --eval-all` but don't want to spend time checking if the formulae is handled by BrewTestBot, since 80-90% of the formulaes outputed by this command are in the list.
> 
> Just to confirm I understand @eleanordoesntcode: is it that this command will still attempt to check the livecheck status for formulae it later complains about being handled by BrewTestBot and you'd rather they were skipped entirely?
Yes, the command will return formulaes that aren't bumped even though they're autobumped by BrewTestBot for the most part.

**Comment by MikeMcQuaid** [comment]
@eleanordoesntcode Thanks for explaining! I think this would be ideally default behaviour, then.
